libr/fs/p/fs_r2.c

```diff
@@ -79,8 +79,9 @@ static RFSFile* fs_r2_open(RFSRoot *root, const char *path, bool create) {
 	for (i = 0; routes[i].path; i++) {
 		const char *cwd = routes[i].path;
 		if (routes[i].cat && !strncmp (path, cwd, strlen (cwd))) {
-			routes[i].cat (root, NULL, path);
-			return NULL;
+			RFSFile* file = r_fs_file_new (root, path);
+			routes[i].cat (root, file, path);
```

Comment:
Return what cat returns

Reply:
cat returns an int, however fs_r2_open should return an RFSFile